### PR TITLE
Updated ruff and pre-commit to 0.7.0 and 4.0.1 respectively

### PR DESCRIPTION
**Pull request by dpy013**
### Link to issue number:
follow up #17260 
### Summary of the issue:
Updated ruff and pre-commit to 0.7.0 and 4.0.1 respectively.
### Description of user facing changes
none
### Description of development approach
Updated version of ruff and pre-commit in requirements.txt file
### Testing strategy:
Need to check if appveyor was able to build successfully and local is building.
### Known issues with pull request:
none
### Code Review Checklist:

- [ ] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [ ] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [ ] API is compatible with existing add-ons.
- [ ] Security precautions taken.

@coderabbitai summary

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -41,8 +41,8 @@ sphinx==8.1.2
 sphinx_rtd_theme==3.0.1
 
 # Requirements for automated linting
-ruff==0.6.3
-pre-commit==3.7.1
+ruff==0.7.0
+pre-commit==4.0.1
```
Comment (seanbudd) [inline requirements.txt:45]: items need to be added to the change log in changes for developers.
Can you please also include changes in #17260, the changes to the pre-commit YML file.
Please also add the suggest new hooks: check-illegal-windows-names and maybe the fixed requirements-txt-fixer

**Comment by dpy013** [comment]
The first two issues, one of adding the update log, and the issue of applying the .pre-commit-config.yaml configuration item from pr17260 have been resolved.
As for adding the check-illegal-windows-names configuration, I have a bit of a question:
I checked.
https://github.com/pre-commit/pre-commit-hooks/issues/589
Among the mentions:
There are file and directory names which are legal in Git and most operating systems but illegal in Windows. They should be avoided in case a developer happens to use Windows.
So what files are we excluding?

**Comment by seanbudd** [comment]
@dpy013 you would be adding the hook to the pre-commit.yml file, it's up to the hook to determine these things. It should require minimal configuration.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -86,6 +86,8 @@ repos:
             test_speechManager/speechManagerTestHarness.py
           )
         )
+    - id: check-illegal-windows-names  # Added hook
+    - id: requirements-txt-fixer  # Added hook
```
Comment (seanbudd) [inline .pre-commit-config.yaml:90]: can you make these comments more descriptive?
It looks like `requirements-txt-fixer` has messed up the file still, perhaps best to remove it?

Comment (dpy013) [reply .pre-commit-config.yaml:90]: Can you help assist with modifications here?

Comment (seanbudd) [reply .pre-commit-config.yaml]: @dpy013 we can take on this work later

**Comment by seanbudd** [comment]
Closing in favour of #17260

**Pull request closed without merging**